modules/taboolaBidAdapter.js

```diff
@@ -170,15 +170,24 @@ function getSiteProperties({publisherId, bcat = []}, refererInfo) {
 
 function getImps(validBidRequests) {
   return validBidRequests.map((bid, id) => {
-    const {tagId, bidfloor = null, bidfloorcur = CURRENCY} = bid.params;
-
-    return {
+    const {tagId} = bid.params;
+    const imp = {
       id: id + 1,
       banner: getBanners(bid),
-      tagid: tagId,
-      bidfloor,
-      bidfloorcur,
-    };
+      tagid: tagId
+    }
+    if (typeof bid.getFloor === 'function') {
+      const floorInfo = bid.getFloor({
+        currency: CURRENCY,
+        mediaType: BANNER,
+        size: '*'
+      });
+      if (typeof floorInfo === 'object' && floorInfo.currency === CURRENCY && !isNaN(parseFloat(floorInfo.floor))) {
+        imp.bidfloor = parseFloat(floorInfo.floor);
+        imp.bidfloorcur = CURRENCY;
+      }
+    }
+    return imp;
```

Comment:
One thing some adapters have done is fallback to use their bidder specific floor if the getFloor function is not present.

Totally up to you if you want to keep that logic or not.

So basically

```js
if (typeof bid.getFloor === 'function') {
   // get floor from getFloor function
} else {
   // get floor from your bidder params if there
}
```

Reply:
@robertrmartinez 

Thanks for the comment, I added the fallback to bid params as you described - thanks!

Reply:
Sorry for the delay here!

Can you add a unit test or two for this change as well please.